File: problems/simplified_tetris/submissions/accepted/simplified_tetris.c
#include <stdio.h>

int main(int argc, char const *argv[])
{
    int N, S, D;
    scanf("%d", &D);
    scanf("%d", &S);
    scanf("%d", &N);

    if (N == 1)
    {
        printf("%d", S);
    }
    else if ((2 * D + S) >= N)
    {
        if ((N % 2) == 0)
        {
            printf("%d", (4 * D + S) / N);
        }
        else
        {
            if ((2 * D) >= S)
            {
                printf("%d", S);
            }
            else
            {
                printf("%d", (4 * D + S) / N);
            }
        }
    }
    else
    {
        printf(0);
    }

    return 0;
}

File: problems/simplified_tetris/submissions/accepted/simplified_tetris.go
package main

import "fmt"

func main() {
	var d, s, n uint
	fmt.Scanf("%d", &d)
	fmt.Scanf("%d", &s)
	fmt.Scanf("%d", &n)

	if n == 1 {
		fmt.Println(s)
	} else if (2*d + s) >= n {
		if n%2 == 0 {
			fmt.Println((4*d + s) / n)
		} else {
			if 2*d >= s {
				fmt.Println(s)
			} else {
				fmt.Println((4*d + s) / n)
			}
		}
	} else {
		fmt.Println(0)
	}
}
